### PR TITLE
Hijack on ground fix

### LINKED ISSUE #5566: Hijack is preventable by spamming the button
## Testmerges:

Unrelated

## Reproduction:
Title
This bug report is mostly so admins have basis to bonk people doing it

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Hive leader could unlock alamo doors before the 30 sec channeling was finished.

This caused hive leader to stop channeling, making it impossible to change the hijack_state and possible for humans to delay hijack by moving alamo from lz1 to lz2 and vice-versa.

That is now being fixed, and unlocking door now disable remote control; The console in alamo cockpit will still be functional.
Touching the dropship console as hive leader will have the same effect as a normal hijack, even in the hive didn't unlocked the door

## Why It's Good For The Game

fixes #5566 
Ai were abusing that by moving the alamo groundside over and over, while the only solution for hiveleader was to spam quicker than ai Alamo's console. Beside that, unlocking alamo doors will now take 30 seconds for the hive, wich is more fair

## Changelog
:cl:
fix: Xenos unlocking of doors is no longer instant
add: New hijack state (unlocked) triggered when alamo is locked down entirely and hive leader use the call down dropship verb.   
add: Touching the dropship console as hive leader is now the same as calling it down, preventing delaying abuse by AI
tweak : changed the priority announcements a bit so it corresponds more to what's really happening, shortened the length of hijack.ogg
/:cl:

### CODE REVIEW

**Comment by MLGTASTICa** [comment]
This doesn't fix the issue.The proper fix would be to stop allowing AI's to interact with the computer consoles to move the alamo once the hijack counter starts.

**Comment by BeloneX** [comment]
You know, the Queen on CM can just disable the dropships remote control by clicking on the console.

**Comment by BraveMole** [comment]
> This doesn't fix the issue.The proper fix would be to stop allowing AI's to interact with the computer consoles to move the alamo once the hijack counter starts.

if(!do_after(user, 30 SECONDS, FALSE, null, BUSY_ICON_DANGER, BUSY_ICON_DANGER))
			to_chat(user, "<span class='warning'>We cease overriding the shuttle lockdown.</span>")
			return FALSE
		if(!is_ground_level(D.z))
			to_chat(user, "<span class='warning'>The bird has left meanwhile, try again.</span>")
			return FALSE
		D.set_hijack_state(HIJACK_STATE_CALLED_DOWN)
		D.unlock_all()
		D.do_start_hijack_timer(GROUND_LOCKDOWN_TIME)

If you stop channeling, you don't change the hijack_state. The hijack state called down is what remove autopilot, hence preventing the abuse

It does the same thing as calling the alamo when you call it down when it's shipside 

if(hijack_state != HIJACK_STATE_NORMAL)
		return
	unlock_all()
	switch(mode)
		if(SHUTTLE_IDLE)
			set_hijack_state(HIJACK_STATE_CALLED_DOWN)
			request_to(S)


That being said, it means that marine can't lock the door, else they'll be stuck on ground, wich is stupid. That change is probably not what needs to be done

File: code/modules/shuttle/marine_dropship.dm [hunk 1]
```diff
@@ -260,8 +261,10 @@
 
 
 /obj/docking_port/mobile/marine_dropship/getStatusText()
-	if(hijack_state != HIJACK_STATE_NORMAL)
-		return "control integrity compromised"
+	if(hijack_state == HIJACK_STATE_CALLED_DOWN)
+		return "Control integrity compromised"
+	else if(hijack_state==HIJACK_STATE_UNLOCKED)
+		return "Remote control compromised"
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:267]:
```suggestion
	else if(hijack_state == HIJACK_STATE_UNLOCKED)
		return "Remote control compromised"
```

File: code/modules/shuttle/marine_dropship.dm [hunk 2]
```diff
@@ -307,6 +310,8 @@
 		return
 
 	hive?.xeno_message("[src] has summoned down the metal bird to [port], gather to her now!")
+	priority_announce("Unknown entity tampering with dropship control. Shutting down autopilot","Dropship departure from ship")
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:313]:
```suggestion
	priority_announce("Unknown interference with dropship control. Shutting down autopilot",  "Dropship malfunction")
```

File: code/modules/shuttle/marine_dropship.dm [hunk 3]
```diff
@@ -361,13 +365,14 @@
 		if(!is_ground_level(D.z))
 			to_chat(user, "<span class='warning'>The bird has left meanwhile, try again.</span>")
 			return FALSE
-		D.set_hijack_state(HIJACK_STATE_CALLED_DOWN)
+		D.set_hijack_state(HIJACK_STATE_UNLOCKED)
 		D.unlock_all()
 		D.do_start_hijack_timer(GROUND_LOCKDOWN_TIME)
 		to_chat(user, "<span class='warning'>We have overriden the shuttle lockdown!</span>")
 		playsound(user, "alien_roar", 50)
+		priority_announce("Alamo lockdown protocol compromised. Interferences preventing remote control", "Dropship Lock Alert")
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:373]:
```suggestion
		priority_announce("Alamo lockdown protocol compromised. Interference preventing remote control", "Dropship Lock Alert")
```

File: code/modules/shuttle/marine_dropship.dm [hunk 4]
```diff
@@ -436,6 +441,7 @@
 	if(M)
 		dat += "<A href='?src=[REF(src)];hijack=1'>Launch to [SSmapping.configs[SHIP_MAP].map_name]</A><br>"
 		M.unlock_all()
+		M.hijack_state=HIJACK_STATE_CALLED_DOWN
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:444]:
```suggestion
		M.hijack_state = HIJACK_STATE_CALLED_DOWN
```

File: code/modules/shuttle/marine_dropship.dm [hunk 5]
```diff
@@ -620,7 +626,7 @@
 	crashing_dropship.crashing = TRUE
 	crashing_dropship.unlock_all()
 	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_DROPSHIP_HIJACKED)
-	priority_announce("Unscheduled dropship departure detected from operational area. Hijack likely. Shutting down autopilot.", "Dropship Alert", sound = 'sound/AI/hijack.ogg')
+	priority_announce("Unscheduled dropship departure detected from operational area. Hijack likely.", "Dropship Alert")
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:629]: Revert this

File: code/modules/shuttle/marine_dropship.dm [hunk 6]
```diff
@@ -968,19 +974,20 @@
...
-		for(var/obj/docking_port/stationary/S in SSshuttle.stationary)
-			if(!options.Find(S.id))
-				continue
-			if(!M.check_dock(S, silent=TRUE))
-				continue
-			destination_found = TRUE
-			dat += "<A href='?src=[REF(src)];move=[S.id]'>Send to [S.name]</A><br>"
-		if(!destination_found)
-			dat += "<B>Shuttle Locked</B><br>"
+		if (M.hijack_state==HIJACK_STATE_NORMAL)
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:980]: ?.
and lacking spaces

File: code/modules/shuttle/marine_dropship.dm [hunk 7]
```diff
@@ -539,7 +545,7 @@
 	var/obj/docking_port/mobile/marine_dropship/M = SSshuttle.getShuttle(shuttleId)
 	if(!M)
 		return
-	if(M.hijack_state != HIJACK_STATE_NORMAL)
+	if(M.hijack_state == HIJACK_STATE_CALLED_DOWN)
 		return
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:549]: Why?

File: code/modules/shuttle/marine_dropship.dm [hunk 8]
```diff
@@ -563,7 +569,7 @@
 	var/obj/docking_port/mobile/marine_dropship/M = SSshuttle.getShuttle(shuttleId)
 	if(!M)
 		return
-	if(!isxeno(usr) && M.hijack_state != HIJACK_STATE_NORMAL)
+	if(!isxeno(usr) && M.hijack_state == HIJACK_STATE_CALLED_DOWN)
 		to_chat(usr, "<span class='warning'>The shuttle isn't responding to commands.</span>")
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:573]: Why?

File: code/modules/shuttle/marine_dropship.dm [hunk 9]
```diff
@@ -272,10 +275,10 @@
 	return ..()
 
 
-/mob/living/carbon/xenomorph/proc/calldown_dropship()
+/mob/living/carbon/xenomorph/proc/hijack()
 	set category = "Alien"
-	set name = "Call Down Dropship"
-	set desc = "Call down the dropship to the closest LZ"
+	set name = "Hijack"
```
Comment (TiviPlus) [inline code/modules/shuttle/marine_dropship.dm:280]:
```suggestion
	set name = "Hijack Dropship"
```

File: code/modules/shuttle/marine_dropship.dm (hunk 5 above)
Comment (BraveMole) [reply code/modules/shuttle/marine_dropship.dm:629]: We already say that the autopilot was shut down in a prior annoncment (when shrike call down alamo). Would better to change the sound, maybe just removing the last sentence

File: code/modules/shuttle/marine_dropship.dm (hunk 7 above)
Comment (BraveMole) [reply code/modules/shuttle/marine_dropship.dm:549]: we want the pilot to be able to use the consol in cockpit, else the hive leader unlocking the alamo would be a sure death sentence for marines. Here they have a chance to get out

File: code/modules/shuttle/marine_dropship.dm (hunk 6 above)
Comment (BraveMole) [reply code/modules/shuttle/marine_dropship.dm:980]: The console at CIC still proposed destination even when alamo was hijacked, even though clicking on the buttons didn't do anything. It is now more consistent

File: code/modules/shuttle/marine_dropship.dm (hunk 8 above)
Comment (BraveMole) [reply code/modules/shuttle/marine_dropship.dm:573]: Same as above. I suspect this code is not reacheable though

File: code/modules/shuttle/marine_dropship.dm (hunk 5 above)
Comment (BraveMole) [reply code/modules/shuttle/marine_dropship.dm]: Added back sound

**Review by TiviPlus — APPROVED** [review]
What did you change with the hijack announcement ogg?

**Comment by BraveMole** [comment]
> What did you change with the hijack announcement ogg?

It was 8 seconds long, it's now 6.5 second long. I edited out the autopilot part

**Comment by TiviPlus** [comment]
ok

**Pull request merged**